src/deckardcain.js

```diff
@@ -13,51 +13,36 @@ const LEGACY_BLUEPRINT_TITLE = /\-{3} ([^\n\r]+ )?\-{3}([\n\r]{1,2}|$)/;
  */
 function identify(source) {
   // Stay awhile and listen!
+  if (source.match(API_BLUEPRINT_HEADER)) {
+    // I spotted 'FORMAT: 1A' header, which gives us a clear clue that
+    // the file is API Blueprint.
+    return 'text/vnd.apiblueprint';
+  }
 
-  let identifiedType = null;
-
-  try {
-    const json = JSON.parse(source);
-
-    // It looks like the source is a valid JSON file. I suspect it to
-    // be a Swagger file.
-    if (json.swagger) {
-      // Indeed, we are dealing with Swagger file!
-      identifiedType = 'application/swagger+json';
-    }
-  } catch (jsonException) {
-    // Well ok, it's not a JSON...Maybe we're dealing with YAML file?
-    try {
-      const yaml = YAML.safeLoad(source);
-
-      if (yaml.swagger) {
-        // Indeed, we are dealing with Swagger file!
-        identifiedType = 'application/swagger+yaml';
-      }
-    } catch (yamlException) {
-      // Well ok I am sorry, it's not a Swaggerish
-    }
-  } finally {
-    if (identifiedType === null) {
-      if (source.match(API_BLUEPRINT_HEADER)) {
-        // I spotted 'FORMAT: 1A' header, which gives us a clear clue that
-        // the file is API Blueprint.
-        identifiedType = 'text/vnd.apiblueprint';
-      } else if (source.match(LEGACY_BLUEPRINT_TITLE)) {
-        // I spotted '--- Sample Title ---' title. This looks like
-        // we are dealing with the legacy Apiary Blueprint.
-        identifiedType = 'text/vnd.legacyblueprint';
-      } else if (source.match(API_BLUEPRINT_RESPONSE)) {
-        // I can not see the '--- Sample Title ---' and at the same time
-        // there is something like '+ Response 200' in the document, which is
-        // pretty distinctive for API Blueprint.
-        identifiedType = 'text/vnd.apiblueprint';
-      }
-    }
-
-    return identifiedType;
+  if (source.match(SWAGGER_YAML)) {
+    // Indeed, we are dealing with Swagger file!
+    return 'application/swagger+yaml';
+  }
+
+  if (source.match(SWAGGER_JSON)) {
+    // Indeed, we are dealing with Swagger file!
+    return 'application/swagger+json';
   }
-}
 
+  if (source.match(LEGACY_BLUEPRINT_TITLE)) {
+    // I spotted '--- Sample Title ---' title. This looks like
+    // we are dealing with the legacy Apiary Blueprint.
+    return 'text/vnd.legacyblueprint';
+  }
+
+  if (source.match(API_BLUEPRINT_RESPONSE)) {
+    // I can not see the '--- Sample Title ---' and at the same time
+    // there is something like '+ Response 200' in the document, which is
+    // pretty distinctive for API Blueprint.
+    return 'text/vnd.apiblueprint';
+  }
+
+  return null;
```

Comment:
Well the flow of this file now looks almost like one big switch/case. Although I crafted the comments to look like Deckard's speach, I think it's not worth to keep them in that form anymore and we should simplify it so it's more readable:

```js
if (source.match(API_BLUEPRINT_RESPONSE)) {
  // not '--- Sample Title ---' and '+ Response 200'
  return 'text/vnd.apiblueprint';
}
```